src/neuroconv/tools/spikeinterface/spikeinterface.py

```diff
@@ -1436,3 +1464,178 @@ def write_sorting(
+    if nwbfile is not None:
+        assert isinstance(nwbfile, pynwb.NWBFile), "'nwbfile' should be a pynwb.NWBFile object!"
+
+    # retrieve templates and stds
+    template_means = waveform_extractor.get_all_templates()
+    template_stds = waveform_extractor.get_all_templates(mode="std")
+    sorting = waveform_extractor.sorting
+    if unit_ids is not None:
+        unit_indices = sorting.ids_to_indices(unit_ids)
+        template_means = template_means[unit_indices]
+        template_stds = template_stds[unit_indices]
+
+    # metrics properties (quality, template) are temporarily added as properties
+    # to be written in the Units table
+    tmp_properties = []
+    if waveform_extractor.is_extension("quality_metrics"):
+        qm = waveform_extractor.load_extension("quality_metrics").get_data()
+        for prop in qm.columns:
+            if prop not in sorting.get_property_keys():
+                sorting.set_property(prop, qm[prop])
+                tmp_properties.append(prop)
+    if waveform_extractor.is_extension("template_metrics"):
+        tm = waveform_extractor.load_extension("template_metrics").get_data()
+        for prop in tm.columns:
+            if prop not in sorting.get_property_keys():
+                sorting.set_property(prop, tm[prop])
+                tmp_properties.append(prop)
+
+    assert write_as in [
+        "units",
+        "processing",
+    ], f"Argument write_as ({write_as}) should be one of 'units' or 'processing'!"
+    if write_as == "units":
+        assert units_name == "units", "When writing to the nwbfile.units table, the name of the table must be 'units'!"
+    write_in_processing_module = False if write_as == "units" else True
+
+    try:
+        with make_or_load_nwbfile(
+            nwbfile_path=nwbfile_path, nwbfile=nwbfile, metadata=metadata, overwrite=overwrite, verbose=verbose
+        ) as nwbfile_out:
+            if write_electrical_series:
+                if not waveform_extractor_has_recording(waveform_extractor):
+                    assert recording is not None, (
+                        "To write electrical series, the waveform_extractor needs to have a recording attached. "
+                        "For recordingless mode, you can use the 'recording' argument."
+                    )
+                else:
+                    recording = waveform_extractor.recording
+                if write_electrical_series_kwargs is None:
+                    write_electrical_series_kwargs = {}
+                write_recording(
+                    recording=recording, nwbfile=nwbfile_out, metadata=metadata, **write_electrical_series_kwargs
+                )
+            else:
+                # we need add_electrodes_from_waveforms
+                if not waveform_extractor_has_recording(waveform_extractor):
+                    if recording is not None:
+                        assert isinstance(recording, BaseRecording), "'recording' needs to be a BaseRecording object."
```

Comment:
So silly and petty, I know, but you're SO close to 100% coverage on the new code additions for this PR - can you add a test for this assertion?

Reply:
This block ends in the part that goes to `write_electrical_series`. I think that that is somwerhere deeper in the stack (probably in `add_electrical_series`) where the responsible for checking correct inputs should be checked.

Reply:
sure!

Reply:
I agree. Removing the assertion and using the `add_electrical_series` function instead of the `write_*`